core/src/Threads/Kokkos_Threads_Instance.cpp

```diff
@@ -255,7 +256,8 @@ int ThreadsInternal::in_parallel() {
 }
 void ThreadsInternal::fence() {
   fence("Kokkos::ThreadsInternal::fence: Unnamed Instance Fence");
-}
+#endif
+}  // namespace Impl
```

Comment:
The last two lines have to be above the first to lines (and the comment is wrong).